### PR TITLE
Build akka-persistence-query on scala3

### PR DESCRIPTION
**Pull request by raboof**
#30243

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM with some questions

File: akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/AllPersistenceIdsStage.scala [hunk 1]
```diff
@@ -30,6 +30,8 @@ final private[akka] class AllPersistenceIdsStage(liveQuery: Boolean, writeJourna
 
   override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
     new TimerGraphStageLogicWithLogging(shape) with OutHandler with Buffer[String] {
+      override def doPush(out: Outlet[String], elem: String): Unit = super.push(out, elem)
```
[the remark this answers is not in the document]
Comment (raboof) [reply akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/AllPersistenceIdsStage.scala:33]: `push` is protected in `GraphStageLogic`.

It looks like previously this made it available in `Buffer { self: GraphStageLogic =>`, but not anymore.

[the remark this answers is not in the document]
Comment (johanandren) [reply akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/AllPersistenceIdsStage.scala:33]: Thx for the clarification!

File: akka-persistence/src/test/scala/akka/persistence/EndToEndEventAdapterSpec.scala [hunk 2]
```diff
@@ -181,7 +181,7 @@ class EndToEndEventAdapterSpec extends AnyWordSpecLike with Matchers with Before
 
     "use the same adapter when reading as was used when writing to the journal" in
     withActorSystem("SimpleSystem", adaptersConfig) { implicit system =>
-      val p = TestProbe()
+      val p: TestProbe = TestProbe()
```
[the remark this answers is not in the document]
Comment (raboof) [reply akka-persistence/src/test/scala/akka/persistence/EndToEndEventAdapterSpec.scala]: The compiler claims so:
```
[error] -- [E045] Cyclic Error: /home/aengelen/dev/akka/akka-persistence/src/test/scala/akka/persistence/EndToEndEventAdapterSpec.scala:185:25 
[error] 185 |      implicit val ref = p.ref
[error]     |                         ^
[error]     |                         Recursive value p needs type
```
though I don't see why it considers `p` recursive. Adding an explicit type to the `implicit val` makes the compiler happy as well, and is neater. I'll update.

(tried to create a small reproducer of the seemingly-incorrect error message but didn't manage to)

**Pull request merged**